feat: Adding support for depreacted both on args and input

lpessoa opened the pull request:
<!--
First, 🌠 thank you 🌠 for taking the time to consider a contribution to Apollo!

Here are some important details to follow:

* ⏰ Your time is important
        To save your precious time, if the contribution you are making will
        take more than an hour, please make sure it has been discussed in an
        issue first. This is especially true for feature requests!

* 💡 Features
        Feature requests can be created and discussed within a GitHub Issue.
        Be sure to search for existing feature requests (and related issues!)
        prior to opening a new request. If an existing issue covers the need,
        please upvote that issue by using the 👍 emote, rather than opening a
        new issue.

* 🕷 Bug fixes
        These can be created and discussed in this repository. When fixing a bug,
        please _try_ to add a test which verifies the fix.  If you cannot, you should
        still submit the PR but we may still ask you (and help you!) to create a test.

* 📖 Contribution guidelines
        Follow https://github.com/apollographql/rover/blob/HEAD/CONTRIBUTING.md
        when submitting a pull request.  Make sure existing tests still pass, and add
        tests for all new behavior.

* ✏️ Explain your pull request
        Describe the big picture of your changes here to communicate to what
        your pull request is meant to accomplish. Provide 🔗 links 🔗 to
        associated issues!

We hope you will find this to be a positive experience! Open source
contribution can be intimidating and we hope to alleviate that pain as much
as possible. Without following these guidelines, you may be missing context
that can help you succeed with your contribution, which is why we encourage
discussion first. Ultimately, there is no guarantee that we will be able to
merge your pull-request, but by following these guidelines we can try to
avoid disappointment.

-->

Fixing an issue where the rover will remove deprecated input fields when using the `rover graph introspect` command.



It requires the following proposed changes to `graphql-client` crate https://github.com/graphql-rust/graphql-client/pull/553.

lpessoa committed:
feat: Addig support for depreacted both on args and input

apollo-cla commented:
@lpessoa: Thank you for submitting a pull request! Before we can merge it, you'll need to sign the Apollo Contributor License Agreement here: https://contribute.apollographql.com/

lpessoa committed:
tests: Updating fixtures for simple and interfaces schema tests

Updating fixture according to changes in introspection query and graphql_client crate

lpessoa committed:
tests: Updating fixtures for swapi tests

lpessoa committed:
Merge branch 'main' into main

lpessoa commented on crates/rover-client/src/operations/graph/introspect/schema.rs:
```diff
@@ -244,26 +244,32 @@ impl Schema {
             value_def.description(desc);
         }
+
+        if value.is_deprecated {
```
I will add them to the fixtures

lpessoa commented on crates/rover-client/src/operations/graph/introspect/fixtures/swapi.json:
```diff
@@ -5950,7 +6160,9 @@
                             },
-                            "defaultValue": "\"No longer supported\""
+                            "defaultValue": "\"No longer supported\"",
+                            "isDeprecated": false,
```
Will update this one to include a arg/input deprecated scenario
